tests/src/whisk/core/controller/test/RejectBigEntitesTests.scala

```diff
@@ -0,0 +1,226 @@
+    val parametersLimit = config.entityParametersLimit
+
+    it should "allow get requests on actions" in {
+        implicit val tid = transid()
+        Get("/api/v1/actions") ~> sealRoute(routes) ~> check {
+            status should be(OK)
+        }
+    }
+
+    it should "allow get requests on packages" in {
+        implicit val tid = transid()
+        Get("/api/v1/packages") ~> sealRoute(routes) ~> check {
+            status should be(OK)
+        }
+    }
+
+    it should "allow post without body requests on actions" in {
+        implicit val tid = transid()
+        Post("/api/v1/actions") ~> sealRoute(routes) ~> check {
+            status should be(OK)
+        }
+    }
+
+    it should "allow post without body requests on packages" in {
+        implicit val tid = transid()
+        Post("/api/v1/packages") ~> sealRoute(routes) ~> check {
+            status should be(OK)
+        }
+    }
+
+    it should "allow post requests on packages with parameters which are too big" in {
+        implicit val tid = transid()
+        val keys: List[Long] = List.range(Math.pow(10, 9) toLong, (parametersLimit.toBytes / 2 / 20 + Math.pow(10, 9) + 2) toLong)
+        val parameters = keys map { key =>
+            Parameters(key.toString, "a" * 10)
+        } reduce (_ ++ _)
+        val content = s"""{"exec":{"kind":"nodejs","code":"??"},"annotations":$parameters}""".stripMargin.parseJson.asJsObject
+        Post("/api/v1/packages", content) ~> sealRoute(routes) ~> check {
+            status should be(OK)
+        }
+    }
+
+    it should "allow delete requests on actions" in {
+        implicit val tid = transid()
+        Delete("/api/v1/actions") ~> sealRoute(routes) ~> check {
+            status should be(OK)
+        }
+    }
+
+    it should "allow delete requests on packages" in {
+        implicit val tid = transid()
+        Delete("/api/v1/packages") ~> sealRoute(routes) ~> check {
+            status should be(OK)
+        }
+    }
+
+    it should "allow request with exec, parameters and annotations" in {
+        implicit val tid = transid()
+        val content = s"""{"exec":{"kind":"python","code":"???"}, "parameters": [{"key": "a", "value": "a"}, {"key": "b", "value": "b"}], "annotations": [{"key": "a", "value": "a"}, {"key": "b", "value": "b"}]}""".stripMargin.parseJson.asJsObject
+        Put(s"/api/v1/actions", content) ~> sealRoute(routes) ~> check {
+            status should be(OK)
+        }
+    }
+
+    it should "reject create action with exec which is too big" in {
+        implicit val tid = transid()
+        val code = "a" * (actionLimit.toBytes / 2L).toInt + 1
+        val content = s"""{"exec":{"kind":"python","code":"$code"}}""".stripMargin.parseJson.asJsObject
+        Put(s"/api/v1/actions", content) ~> sealRoute(routes) ~> check {
+            status should be(RequestEntityTooLarge)
+            response.entity.toString should include("request entitiy too large")
```

Comment:
typo repeated several times - list this to a const string (since this message may change).